pinot-core/src/test/java/com/linkedin/pinot/core/periodictask/PeriodicTaskSchedulerTest.java

```diff
@@ -0,0 +1,181 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.periodictask;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class PeriodicTaskSchedulerTest {
+
+  @Test
+  public void testSchedulerWithOneTask() {
+    AtomicInteger count = new AtomicInteger(0);
+    PeriodicTaskScheduler periodicTaskScheduler = new PeriodicTaskScheduler(0L);
+    long runFrequencyInSeconds = 1L;
+    long totalRunTimeInMilliseconds = 4_000L;
+
+    List<PeriodicTask> periodicTasks = new ArrayList<>();
+    PeriodicTask task = new BasePeriodicTask("Task", runFrequencyInSeconds, 0L) {
+      @Override
+      public void run() {
+        // Execute task.
+        count.incrementAndGet();
+      }
+    };
+    periodicTasks.add(task);
+
+    long start = System.currentTimeMillis();
+    periodicTaskScheduler.start(periodicTasks);
+    try {
+      Thread.sleep(totalRunTimeInMilliseconds);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    }
+    periodicTaskScheduler.stop();
+
+    Assert.assertTrue(count.get() > 0);
+    Assert.assertTrue(count.get() < (totalRunTimeInMilliseconds / runFrequencyInSeconds));
+    Assert.assertTrue(totalRunTimeInMilliseconds < (System.currentTimeMillis() - start));
+  }
+
+  @Test
+  public void testSchedulerWithTwoStaggeredTasks() {
+    AtomicInteger count = new AtomicInteger(0);
+    PeriodicTaskScheduler periodicTaskScheduler = new PeriodicTaskScheduler(0L);
+    long runFrequencyInSeconds = 4L;
+    long totalRunTimeInMilliseconds = 15_000L;
+
+    List<PeriodicTask> periodicTasks = new ArrayList<>();
+    PeriodicTask task1 = new BasePeriodicTask("Task1", runFrequencyInSeconds, 0L) {
+      @Override
+      public void run() {
+        // Execute task.
+        count.incrementAndGet();
+      }
+    };
+    periodicTasks.add(task1);
+
+    // Stagger 2 tasks.
+    PeriodicTask task2 = new BasePeriodicTask("Task2", runFrequencyInSeconds, runFrequencyInSeconds / 2) {
+      @Override
+      public void run() {
+        // Execute task.
+        count.decrementAndGet();
+      }
+    };
+    periodicTasks.add(task2);
+
+    long start = System.currentTimeMillis();
+    periodicTaskScheduler.start(periodicTasks);
+    try {
+      Thread.sleep(totalRunTimeInMilliseconds);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```

Comment:
I am not sure catching interrupted exception is a good idea. You can add throws to the method signature and let the test fail and bail instead of continue on.